src/lcm/simulation/simulate.py

```diff
@@ -89,8 +85,29 @@ def simulate(
     # ----------------------------------------------------------------------------------
     simulation_results = {}
 
+    last_period = n_periods - 1
+
     for period in range(n_periods):
-        state_action_space = state_action_space.replace(states)
+        logger.info("Period: %s", period)
+
+        is_last_period = period == last_period
+
+        relevant_states = model.variable_info.query(
+            "is_state and enters_concurrent_valuation"
+        ).index.tolist()
+        if not is_last_period:
+            relevant_states += model.variable_info.query(
+                "is_state and enters_transition"
+            ).index.tolist()
+
+        states_for_state_action_space = {n: states[n] for n in set(relevant_states)}
```

Comment:
```suggestion
        if is_last_period:
            query = "is_state and enters_concurrent_valuation"
        else:
            query = "is_state and (enters_concurrent_valuation | enters_transition)"
        states_for_state_action_space = {
            n: states[n] for n in model.variable_info.query(query).index
        }
```

Seems clearer I hope?